receiver/awslambdareceiver/README.md

```diff
@@ -217,7 +217,70 @@ The Lambda function requires the following IAM permissions:
 - Error retrying 
 
   Error retrying can be configured through the Lambda deployment setting.
-  For example, errors can be forwarded to a S3 bucket.
   Read more about at [AWS error handling for asynchronous invocations](https://docs.aws.amazon.com/lambda/latest/dg/invocation-async-configuring.html).
 
+- Retaining failed records
+
+  This component supports replaying retained failure records stored at S3.
+  Read more about retaining records at [Capture records of Lambda Async Invocations](https://docs.aws.amazon.com/lambda/latest/dg/invocation-async-retain-records.html).
+
+### Error replaying from S3
+
+When an S3 bucket is configured as the destination for retaining failed Lambda records, the receiver supports replaying those failed events for reprocessing.
+To enable this feature, set the `failure_bucket_arn` configuration to the ARN of your S3 bucket used as the Lambda failure destination.
+
+```yaml
+receivers:
+  awslambda:
+    s3:
+      encoding: awslogs_encoding
+    failure_bucket_arn: "arn:aws:s3:::example"
+```
+
+With required configuration present, receiver accepts a custom event to trigger replaying failed events.
+Consider the event structure below,
+
+```json
+{
+  "replayFailedEvents": {
+    "dryrun": false,
+    "removeOnSuccess": true
+  }
+}
+```
+
+JSON key `replayFailedEvents` defines the custom event type for replaying failed events.
+The table below explains supported options,
+
+| Option          | Description                                                                                    | Default |
+|-----------------|------------------------------------------------------------------------------------------------|---------|
+| dryrun          | Run the command without processing. Useful to understand details about replaying error files   | false   |
+| removeOnSuccess | Configure whether to remove error event from S3 error destination, if processing is successful | true    |
+
+> [!NOTE]  
+> It is recommended to use "dryrun" mode to validate the number of replayable errors in the error destination bucket.
+> If there are many errors, Lambda innovation may time out before processing all error entries.
+> If a timeout occur, you will need to run the custom event multiple times to fully process all error events from the bucket.
+
+### Running with AWS CLI
+
+First, you need the name of the deployed Lambda. You can extract this from the deployment.
+Then, invoke the Lambda with command,
+
+```shell
+aws lambda invoke \
+  --function-name <LAMBDA_DEPLOYMENT_NAME> \
+  --payload '{ "replayFailedEvents": {"dryrun": false}}' \
```

Comment:
```suggestion
  --payload '{ "replayFailedEvents": {}}' \
```

Isn't dryrun false by default?

Reply:
Yup it's `false` by default (see default constructor at https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/45042/changes#diff-95c7ce18ec755aaf51f849f03346fbd70ae179a26b887624804409cf89b15a5cR45-R50). I just kept this to show how user can set configurations of the trigger. But we can remove this if you think it's unnecessary.

Reply:
I see. I expect users will come to the docs and copy & paste this though.

I'd suggest minimising it, and perhaps add another example with dryrun set to true, to show how to override the defaults. WDYT?

Reply:
Good point, will add this suggestion